### ISSUE TITLE
cannot use tune_model with a model not in models()

### ISSUE THREAD

**Issue by bgalvao** [pycaret/pycaret#997]
Code sample:

```python
from sklearn.svm import NuSVC

my_svm = create_model(
    NuSVC(kernel='poly', degree=2, probability=True),
#     fit_kwargs={'sample_weight': sample_weight}
)

my_svm_param_grid = {
    'degree': [2, 3],
    'nu': [.2, .5, .8],
    'coef0': [.0, .5, 1.0],
    'class_weight': [{0: 1.0, 1: 1.0}, {0: 1.0, 1: 2.0}]
}

tune_model(my_svm, custom_grid=my_svm_param_grid)
```

Stack trace:

```stderr
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-20-170f5a0ed4c2> in <module>()
----> 1 tune_model(my_svm, custom_grid=my_svm_param_grid)

/home/bee/.anaconda3/envs/dy/lib/python3.7/site-packages/pycaret/classification.py in tune_model(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, **kwargs)
   1099         verbose=verbose,
   1100         tuner_verbose=tuner_verbose,
-> 1101         **kwargs,
   1102     )
   1103 

/home/bee/.anaconda3/envs/dy/lib/python3.7/site-packages/pycaret/internal/tabular.py in tune_model_supervised(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, display, **kwargs)
   4020     estimator_id = _get_model_id(base_estimator)
   4021 
-> 4022     estimator_definition = _all_models_internal[estimator_id]
   4023     estimator_name = estimator_definition.name
   4024     logger.info(f"Base model : {estimator_name}")

KeyError: None
```

I have not yet looked into the source code, but I will speculate that this looks like that PyCaret is looking for the model within its own model library and that, since `my_svm` does not belong there, it errors out.

Is this a possible use case with PyCaret?

**Comment by Yard1** [comment]
We are aware of this issue. It will be fixed in the next release (hopefully).

**Comment by bgalvao** [comment]
As I have professional needs/interest on this issue, I am open to working on it. I would just like some pointers on what I have to change, if that is possible.

**Comment by Yard1** [comment]
We already have made the necessary changes on a branch targeted for the next major release - however, I can port them over to the current branch and we can release a small update soon. Does that work for you?

**Comment by bgalvao** [comment]
yes 🙂

**Comment by Yard1** [comment]
Fixed in commit 73339b016c643f8ea308a7bcd653135155044f94
